### DISCUSSION TITLE
When I invoke an action I wrote with an input, the input doesn't seem to be visible to the action

### DISCUSSION THREAD

**Discussion by mbestavros** [community/community#26063] [Actions]
<p>I’m in the process of writing my own action, contained in this repo: <a href="https://github.com/enarx/nightly-features" rel="nofollow noopener">https://github.com/enarx/nightly-features</a></p>
<p>As part of the action’s <a href="https://github.com/enarx/nightly-features/blob/2ed57f0ccc24b523b10e0f8fee0ad5e40cfcb2fb/action.yml#L8" rel="nofollow noopener">definition</a>, I expect some input (I’ve named it <code>allowed_features</code>). From Github Actions’ <a href="https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs" rel="nofollow noopener">documentation</a>, this should be accessible via the environment variable <code>INPUT_ALLOWED_FEATURES</code> when specified by a calling workflow.</p>
<p>However, this doesn’t seem to be working for me. I’ve written a workflow that invokes this action <a href="https://github.com/mbestavros/enarx/blob/21697909ca23b55e3e857f195de41096f1fffa62/.github/workflows/manual.yml" rel="nofollow noopener">here</a>. It provides the input, as far as I can tell, correctly; however, when I run that workflow, the action I wrote doesn’t seem to be able to read from the expected environment variable. You can see that demonstrated in this workflow run: <a href="https://github.com/mbestavros/enarx/runs/1009733602?check_suite_focus=true" rel="nofollow noopener">https://github.com/mbestavros/enarx/runs/1009733602?check_suite_focus=true</a></p>
<p>I’ve dumped the environment variables into the log from the called action, and it doesn’t deem to have the expected <code>INPUT_ALLOWED_FEATURES</code> among them.</p>
<p>Does anyone have any pointers as to how I’m going wrong? I’ve looked at other examples, and everything I’ve looked at suggests I’m doing things correctly.</p>
<p>Thanks in advance!</p>

**Comment by Simran-B** [comment]
<p>For testing purposes, could you add this to your <code>action.yml</code> and see what it prints?</p>
<pre><code class="lang-auto">    - run: |
        echo "inputs.allowed_features: ${{ inputs.allowed_features }}"
        echo "env.INPUT_ALLOWED_FEATURES: ${{ env.INPUT_ALLOWED_FEATURES }}"
      shell: bash
</code></pre>

**Comment by mbestavros** [comment]
<aside class="quote group-Advisory-Council" data-username="Simran-B" data-post="2" data-topic="128421">
<div class="title">
<div class="quote-controls"></div>
 Simran-B:</div>
<blockquote>
<pre><code class="lang-auto">- run: |
        echo "inputs.allowed_features: ${{ inputs.allowed_features }}"
        echo "env.INPUT_ALLOWED_FEATURES: ${{ env.INPUT_ALLOWED_FEATURES }}"
      shell: bash
</code></pre>
</blockquote>
</aside>
<p><a class="mention" href="/simran-b">@Simran-B</a> Fascinating… <code>inputs.allowed_features</code> seems to work! But <code>env.INPUT_ALLOWED_FEATURES</code> is still empty.</p>
<p>How would one access <code>inputs</code> from the shell (or Python)?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/mbestavros">@mbestavros</a>,</p>
<p>In action.yaml, you can use <a href="https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsenv" rel="nofollow noopener">‘set-env’</a> to add ${{ inputs.allowed_features }} to env context, then you can get it from the shell.</p>
<pre><code class="lang-auto">    - run: echo "::set-env name=INPUT_ALLOWED_FEATURES::${{ inputs.allowed_features }}"
      shell: bash
</code></pre>
<p>I forked your action repo and confirmed it works(comment some shell script), please check it for your reference.<br>
</p><aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/weide-zhou/nightly-features/blob/3d5ebb298f664b66a0c6a7e9264229397d0f264b/action.yml" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/weide-zhou/nightly-features/blob/3d5ebb298f664b66a0c6a7e9264229397d0f264b/action.yml" target="_blank" rel="nofollow noopener">weide-zhou/nightly-features/blob/3d5ebb298f664b66a0c6a7e9264229397d0f264b/action.yml</a></h4>
<pre><code class="lang-yml">name: nightly-features
description: Test enabled Rust nightly features against an allowlist

branding:
  icon: zap
  color: gray-dark

inputs:
  allowed_features:
    description: 'Allowed Rust nightly features. Provide a comma-separated list. For example: asm,naked_functions,raw_ref_macros,ptr_offset_from'
    required: true
    
runs:
  using: "composite"
  steps:
    - run: |
        bash -c "env | sort"
        echo -------------
      shell: bash
    - run: echo "::set-env name=INPUT_ALLOWED_FEATURES::${{ inputs.allowed_features }}"
</code></pre>

  This file has been truncated. <a href="https://github.com/weide-zhou/nightly-features/blob/3d5ebb298f664b66a0c6a7e9264229397d0f264b/action.yml" target="_blank" rel="nofollow noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p><img src="https://user-images.githubusercontent.com/52772761/181095044-08366a3f-2c13-41d2-b1ea-82ca097edd2b.png" alt="image" data-base62-sha1="mp2TnfrYhknN9o5fgpwCKvbJqo" width="489" height="105"></p>
<p>Thanks</p>

**Comment by mbestavros** [comment]
<p><a class="mention" href="/simran-b">@Simran-B</a> <a class="mention" href="/weide-zhou">@weide-zhou</a> Thanks so much to both of you for your help! It works like a charm now :slight_smile:</p>
<p>It would be helpful if the Actions documentation was updated with this alternative. I had no idea of this approach from just reading; having this called out would be helpful for future Actions developers.</p>

**Comment by weide-zhou** [comment]
<p>Glad to support :wink:! <a class="mention" href="/mbestavros">@mbestavros</a></p>
<p>According to the <a href="https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions#contacting-support" rel="nofollow noopener">policy</a>, you can raise a feedback ticket in below link to ask for doc update, github product manager will take a review and help to check, thanks.</p>
<p><a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" class="onebox" target="_blank" rel="nofollow noopener">https://support.github.com/contact/feedback?contact[category]=actions</a></p>
